### DISCUSSION TITLE
transfer learning

### DISCUSSION THREAD

**Discussion by sushanthred** [ultralytics/ultralytics#17354] [Ideas]
hello how do i implement transfer learning for yolo v8 
i will be tedious if the model is trained over the combined dataset which includes my new classes data too
so i want to implement using freeze layers which layers are to be frozen ?

**Comment by sushanthred** [comment]
@glenn-jocher

**Reply by glenn-jocher** [reply]
@sushanthred to implement transfer learning with YOLOv8, you can freeze the initial layers of the model to retain learned features and fine-tune the remaining layers on your new dataset. Typically, you would freeze the backbone layers, which are responsible for feature extraction, and train the head layers that are more specific to your task. If you need further assistance, feel free to ask!

**Reply by sushanthred** [reply]
yeah i tried freezing 10 layers but the result is not as expected so i freeze 53 layers and the result is the same .so is there anyway way or a method or function to know the backbone layers in the yolov8 model that i currently loaded?

**Reply by sushanthred** [reply]
@glenn-jocher

**Reply by glenn-jocher** [reply]
To identify the backbone layers in your YOLOv8 model, you can inspect the model architecture using PyTorch's `named_children()` method, which will help you determine which layers to freeze. If you need further guidance, feel free to ask!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sushanthred, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a custom training ❓ Question, please provide as much information as possible, including details about your dataset, any training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 

Regarding implementing transfer learning for YOLOv8, particularly freezing layers, I suggest experimenting with different layers depending on which features you want the model to retain or relearn. The layers to freeze can vary based on your specific use case and the nature of your new data classes. Additionally, an Ultralytics engineer will assist you soon to provide more detailed guidance 😊.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.
